graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/AggregationSpecToPivotMapper.java

```diff
@@ -47,26 +51,43 @@ public AggregationSpecToPivotMapper(final GroupingToBucketSpecMapper rowGroupCre
 
     @Override
     public Pivot apply(final AggregationRequestSpec aggregationSpec) {
+        final List<BucketSpec> groups = aggregationSpec.groupings()
+                .stream()
+                .map(rowGroupCreator)
+                .collect(Collectors.toList());
+
+        final Map<Metric, SeriesSpec> series = aggregationSpec.metrics()
+                .stream()
+                .filter(Objects::nonNull)
+                .collect(Collectors.toMap(Function.identity(), seriesCreator));
```

Comment:
I'd say Map is an overkill... What about using ImmutablePair instead? You are pairing two values rather than creating mapping between them anyway, right?

Reply:
It provides quite a bit of sugar, like keys() or values() that we use elsewhere.

Reply:
You are right, with map I lost the order of metrics and it would cause problems later. Replaced by a list of pairs. Thanks for the tip!